apps/web/components/ui/write-review-form.tsx

```diff
@@ -50,23 +50,76 @@ export default function WriteReviewForm({ onSubmit, previousReview }: WriteRevie
 		content: previousReview?.description || ""
 	});
 
+	useEffect(() => {
+		if (!previousReview) {
+			const savedAcademicYear = localStorage.getItem("reviewFormAcademicYear");
+			const savedRating = localStorage.getItem("reviewFormRating");
+			const savedDescription = localStorage.getItem("reviewFormDescription");
+
+			if (savedAcademicYear) setAcademicYear(savedAcademicYear);
+			if (savedRating) setRating(parseFloat(savedRating));
+			if (savedDescription && markdownEditor) markdownEditor.commands.setContent(savedDescription);
+		}
+	}, [previousReview, markdownEditor]);
+
+	//to save into localstorage
+	useEffect(() => {
+		if (!previousReview) {
+			const savedAcademicYear = localStorage.getItem("reviewFormAcademicYear");
+			const savedRating = localStorage.getItem("reviewFormRating");
+			const savedDescription = localStorage.getItem("reviewFormDescription");
+
+			if (savedAcademicYear) setAcademicYear(savedAcademicYear);
+			if (savedRating) setRating(parseFloat(savedRating));
+			if (savedDescription && markdownEditor) markdownEditor.commands.setContent(savedDescription);
+		}
+	}, [previousReview, markdownEditor]);
+
+	//to save localStorage when state changes
+	useEffect(() => {
+		localStorage.setItem("reviewFormRating", rating.toString());
+	}, [rating]);
+
+	useEffect(() => {
+		if (academicYear) localStorage.setItem("reviewFormAcademicYear", academicYear);
+		if (markdownEditor) {
+			const saveMarkdown = () => {
+				localStorage.setItem("reviewFormDescription", markdownEditor.storage.markdown.getMarkdown());
```

Comment:
We should really include course code into each key as well. Like:  `reviweFormDescription_${courseCode}}`


This is to prevent the case where:
You write a review for course PSY102, everything is saved and gucci. The problem is if you switch to ITE442 without submitting the review first, whatever you wrote in PSY102 will appear on ITE442 because it is essentially using the same key

Reply:
ohh yess!